llvm/lib/Target/RISCV/AsmParser/RISCVAsmParser.cpp

```diff
@@ -1046,6 +1046,20 @@ struct RISCVOperand final : public MCParsedAsmOperand {
            isInt<26>(fixImmediateForRV32(Imm, isRV64Imm()));
   }
 
+  bool isImm32() const {
+    int64_t Imm;
+    RISCVMCExpr::VariantKind VK = RISCVMCExpr::VK_RISCV_None;
+    if (!isImm())
+      return false;
+    bool IsConstantImm = evaluateConstantImm(getImm(), Imm, VK);
+    bool IsValid;
+    if (!IsConstantImm)
+      IsValid = RISCVAsmParser::classifySymbolRef(getImm(), VK);
+    else
+      IsValid = isInt<32>(Imm) || isUInt<32>(Imm);
```

Comment:
Is this equivalent to `isInt<32>(fixImmediateForRV32(Imm, isRV64Imm()))` like we do for other immediates?

Reply:
Yes this is equivalent, and a neater way to express this. I wrote it this way because i wasn't sure.